src/sdk.spec.js

```diff
@@ -125,17 +125,87 @@ describe('OpenForm', () => {
     expect(form.clientBaseUrl).toEqual('http://localhost/some-subpath');
   });
 
-  it('should correctly set the formUrl (hash fragment routing)', () => {
+  it("shouldn't take basepath into account (hash based routing)", () => {
     mswServer.use(...apiMocks);
-    window.history.pushState({}, 'Dummy title', '/some-server-side/path');
+    window.history.pushState({}, 'Dummy title', '/some-path');
     const formRoot = document.createElement('div');
     const form = new OpenForm(formRoot, {
       baseUrl: BASE_URL,
-      basePath: '/some-subpath/',
       formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
       useHashRouting: true,
     });
 
-    expect(form.clientBaseUrl).toEqual('http://localhost/some-server-side/path#/some-subpath');
+    expect(form.clientBaseUrl).toEqual('http://localhost/');
   });
+
+  it.each([
+    [
+      '/some-subpath?_of_action=afspraak-annuleren',
+      'http://localhost/some-subpath/afspraak-annuleren',
+    ],
+    [
+      '/some-subpath?_of_action=afspraak-maken',
+      'http://localhost/some-subpath/afspraak-maken/producten', // SDK redirects to producten
+    ],
+    [
+      '/some-subpath?_of_action=cosign&_of_action_params=submission_uuid%3Dabc',
+      'http://localhost/some-subpath/cosign?submission_uuid=abc',
+    ],
+    [
+      '/some-subpath?_of_action=stap&_of_action_params=next_step%3Dstep-1',
+      'http://localhost/some-subpath/startpagina', // SDK redirects to start page
+    ],
+  ])('should handle action redirects correctly', async (initialUrl, expected) => {
+    mswServer.use(...apiMocks);
+    const formRoot = document.createElement('div');
+    window.history.pushState(null, '', initialUrl);
+    const form = new OpenForm(formRoot, {
+      baseUrl: BASE_URL,
+      basePath: '/some-subpath',
+      formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
+      lang: 'nl',
+    });
+    await act(async () => await form.init());
+
+    // wait for the loader to be removed when all network requests have completed
+    await waitForElementToBeRemoved(() => within(formRoot).getByRole('status'));
+    expect(location.href).toEqual(expected);
+  });
+
+  it.each([
+    [
+      '/?_of_action=afspraak-annuleren&unrelated_q=1',
+      'http://localhost/?unrelated_q=1#/afspraak-annuleren',
+    ],
+    [
+      '/?_of_action=afspraak-maken&unrelated_q=1',
+      'http://localhost/?unrelated_q=1#/afspraak-maken/producten', // SDK redirects to producten
+    ],
+    [
+      '/?_of_action=cosign&_of_action_params=submission_uuid%3Dabc&unrelated_q=1',
+      'http://localhost/?unrelated_q=1#/cosign?submission_uuid=abc',
+    ],
+    [
+      '/?_of_action=stap&_of_action_params=next_step%3Dstep-1&unrelated_q=1',
+      'http://localhost/?unrelated_q=1#/startpagina', // SDK redirects to start page
```

Comment:
that's odd, shouldn't it go to `...#/stap/step-1`?

Reply:
I suspect the api mocks to not cover an api call, so it goes back to the start page as it doesn't know about `step-1` but I'm not sure about that

Reply:
Edit, figured out while adding tests:
https://github.com/open-formulieren/open-forms-sdk/blob/c87f004fbaa2db868d30260aef648d0d3e5b3aeb/src/sdk.spec.js#L177-L212

The behavior is different depending on having a basepath or not with hash routing, something might be going wrong

Reply:
please investigate further what's causing this :grimacing:

Reply:
so the resume flow relies on the `submission_id` (query) param, as that gets loaded from the backend first before the actual step is rendered. Many of these components are wrapped in a `<RequiredSubmission>` component/decorator.

Probably best to manually test the behaviour with a real submission, it doesn't make sense to redirect to a form/submission that's not in your session anyway.

Reply:
This ended up being a bit more complicated:
We have to differentiate between the react router base path and the "real" browser base path with hash based routing. In hash based routing, router base path will always be '', but we need the extra `browserBasePath` to be able to correctly set `this.clientBaseUrl`, as the CMS basepath needs to be taken into account